Splitting etter postnummer

noenandre opened the pull request:
Nå støtter city_subdivision.py deling etter postnummer. Både kommenummer, kommunenavn og filnavn støttes som argument fra kommandolinja. Bbox er tatt i bruk. Splitting kjøres automatisk fra building2osm.py med flagget -s eller --split.

"city_subdivision.py" er kanskje et dårlig navn på scriptet når det ikke bare er for byer og bydeler lengre?

noenandre committed:
Sjekker bbox før "inside_linear_ring"

noenandre committed:
Bruker pairwise implementasjonen fra more-itertools prosjektet. Egen itertools.pairwise funksjon kommer med Python 3.10.

noenandre committed:
Støtter nå deling etter postnummer. Støtter også kommunenummer, kommunenavn eller filnavn fra bulding2osm fra kommandolinja.

noenandre committed:
Kjører "city_subdivisions" fra "building2osm" med subprocess.run().

noenandre committed:
Merge branch 'bydel_speedups' into postcode_split

NKAmapper commented:
Veldig bra. Dette blir nyttig.

Noen mindre forslag:

- Kunne du bruke xml.etree i stedet for lxml (folk slipper å installere lxml)?
- "bydel" er kanskje bedre enn "bydeler" i filnavnet
- Kunne du legge til print av antallet bygninger per fil (slik får vi tall til progress-wikien)?
- Foreslår bare "-split" som argument i building2osm.py, slik som de andre argumentene der (men ditt forslag fulgte konvensjonene bedre :)
- "_" for delte kommunenavn i filen (Vestre_Slidre etc)
- Flere kommentarer i koden, når du får tid en gang

Navn: Tja, har ikke noe ferdig forslag. Programmet kan egentlig generaliseres en gang i fremtiden til å dele hva som helst av objekter for en kommune. Da passer kanskje municipality_split.py, eller split2osm.py eller noe sånt.

noenandre committed:
Fikset kommalinjeargumenter.
→«-split» istedenfor «-s» og «--split» fra buldings2osm.
→«bydel» istedenfor «bydeler», slik at bydel og postnummer begge blir i ubestemt entall i city_subdivisions.
«bydeler» og «postnummere» i flertall i filnavn for område-polygoner.
Lagt til telling av antall bygninger.

noenandre committed:
Importerer xml.etree.ElementTree som reserve til lxml.
Løst problem med at xml.etree ikke har noen .nsmap attributt.

noenandre committed:
Forenklet signatur på 2 funksjoner.

noenandre committed:
Endret navn på  "city_subdivisions.py" til "municipality_split".

noenandre commented:
Grunnen til at jeg gikk for lxml var at jeg trodde bs4 var avhengig av lxml, slik at folk allerede måtte installere det. Der trodde jeg feil. Nå skal alt virke med xml.etree også, den lastes inn som en reserve. 

Ellers har jeg fulgt dine anbefalinger.

noenandre committed:
Fikset test og rettet navn.

NKAmapper committed:
Simpler print

NKAmapper commented:
Suverent!
Tillot meg å rydde litt i printen. Om du er ok, og om dette er klart nå så kan jeg flette inn dette?

noenandre commented:
Alt er klart til å flettes inn!

NKAmapper commented:
Da er den flettet inn.
Kanskje du poster et hint i forum-tråden?
